### ISSUE TITLE
syntax-begin-function obsolete in Emacs 25.1

### ISSUE THREAD

**Issue by Parent5446** [emacs-typescript/typescript.el#17]
As of Emacs 25.1 the `syntax-begin-function` variable is obsolete. I am not sure what the replacement is.

Here is the compiler warning received:

```
typescript-mode.el:2052:10:Warning:
    `syntax-begin-function' is an obsolete variable (as of 25.1).
```

**Comment by bsed** [comment]
i have same warning

**Comment by josteink** [comment]
I'd hate to use anything PHP as a reference for this, but it seems php-mode has encountered the same issue and had a ... fairly pragmatical approach to it: https://github.com/ejmr/php-mode/pull/340

I guess we *could* do the same, since we too have to support older version of Emacs.

That said, what I find most annoying about this is that I cannot anywhere find any documentation on what this is deprecated *in favour of*. Basically. What should we replace it with? Anyone have any idea?

If there's a reasonable (and backwards compatible) thing we can do instead, I'm all for it.

If not, I say we ignore the warning until it causes actual errors.

**Comment by josteink** [comment]
Note: This is incidentally fixed in PR #45.

**Comment by lddubeau** [comment]
The warning about `syntax-begin-function` came up recently in #102 There's nothing to be done from the perspective of maintaining functionality. `syntax-begin-function` is just no longer used, with no replacement. Setting it in newer Emacs causes a warning but is not otherwise harmful. I'm closing this in favor of #102 where we've been focusing on how to suppress the warning, which is the only issue that remains really.

(Historical note: it turns out that PR #45 was never merged and so has no bearing on how we're using `syntax-begin-function`.)
